doc/source/ray-core/internals/rpc-fault-tolerance.rst

```diff
@@ -0,0 +1,211 @@
+.. _rpc-fault-tolerance:
+
+RPC Fault Tolerance
+===================
+
+All RPCs added to Ray Core should be fault tolerant and use the retryable gRPC client. 
+Ideally, they should be idempotent, or at the very least, the lack of idempotency should be 
```

Comment:
recommend giving a trivial positive and negative example here of something that is and isn't idempotent (people might not be familiar with the term, and if they are might not be able to easily connect it to code examples)

Reply:
like a very dumbed down RPC handler that just adds something to a data structure. the non-idempotent version could append while the idempotent one first checks if it's present. (or whatever you see fit)